addons/scopes/functions/fnc_inventoryCheck.sqf

```diff
@@ -131,10 +127,34 @@ private _newGuns = [primaryWeapon _player, secondaryWeapon _player, handgunWeapo
             GVAR(canAdjustElevation) set [_x, (_verticalIncrement > 0) && !(_maxVertical isEqualTo [0, 0])];
             GVAR(canAdjustWindage) set [_x, (_horizontalIncrement > 0) && !(_maxHorizontal isEqualTo [0, 0])];
         };
-    }
+
+        if (GVAR(enabled) == 1) then {
+            if ((GVAR(canAdjustElevation) select _forEachIndex) || (GVAR(canAdjustWindage) select _forEachIndex)) then {
+                if (!((_adjustment select _forEachIndex) isEqualTo [0,0,0])) then {
+                    // The optic for this weapon changed, set adjustment to zero
+                    _adjustment set [_forEachIndex, [0, 0, 0]];
+                    _updateAdjustment = true;
+                };
+            } else {
+                if (!((_adjustment select _forEachIndex) isEqualTo [])) then {
+                    // Can't change this optic, set to empty set
+                    _adjustment set [_forEachIndex, []];
```

Comment:
I like the idea, but I think it will break the code in other places:
https://github.com/acemod/ACE3/blob/98da86b74d4089bc3403d0b0c0074e6a46ce35d8/addons/scopes/functions/fnc_adjustScope.sqf#L50
https://github.com/acemod/ACE3/blob/98da86b74d4089bc3403d0b0c0074e6a46ce35d8/addons/scopes/functions/fnc_applyScopeAdjustment.sqf#L34